### PR TITLE
283 create api endpointserver func to delete allowed logins

### PR DESCRIPTION
**Pull request by ethanhmaness**
# Description
Added `removeAllowedEmail()` in `src/server/actions/Setting.ts` which removes an email from the allowed emails list in the settings document and deletes a user if there is a user associated with that email.
Added API endpoint at DELETE `/api/settings/emails` and associated zod types to call `removeAllowedEmail()`.

## Relevant issue(s)

#283 

## Questions
Should this include authorization?
Do we want to move this to `Settings.ts`?

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have assigned reviewers to this PR

## To test

Log in as an admin and hit DELETE `/api/settings/emails?email=<email to delete>`.

### CODE REVIEW

File: src/server/actions/Setting.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    } else {
+        // Modifies array in-place
+        doc.allowedEmails.splice(index, 1)
+        // Set database obj array to modified array of retrieved document
+        await SettingsSchema.updateOne({ _id: doc._id }, { allowedEmails: doc.allowedEmails })
+
+        // Delete associated user if one exists
+        try {
+            const user: UserResponse = await getUserByEmail(email)
+            await UserSchema.findOneAndRemove({ _id: user._id })
```
Comment (zaviermiller) [inline src/server/actions/Setting.ts:32]: why make the extra call here? You could just `UserSchema.findOneAndRemove({ email })`

**Review by zaviermiller — CHANGES_REQUESTED** [review]
quick fix

File: src/server/actions/Setting.ts (hunk 1 above)
Comment (ethanhmaness) [reply src/server/actions/Setting.ts:32]: i forgor user emails are unique

File: src/server/actions/Setting.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    } else {
+        // Modifies array in-place
+        doc.allowedEmails.splice(index, 1)
+        // Set database obj array to modified array of retrieved document
+        await SettingsSchema.updateOne({ _id: doc._id }, { allowedEmails: doc.allowedEmails })
+
+        // Delete associated user if one exists
+        try {
+            await UserSchema.findOneAndRemove({ email: email })
+        } catch (e) { }
```
Comment (zaviermiller) [inline src/server/actions/Setting.ts:30]: why are there two try catches here?

Comment (ethanhmaness) [reply src/server/actions/Setting.ts:30]: First try catch is for throwing error if database call fails. Second try-catch never throws an error, it's just necessary to delete a user with the given email if one exists and to do nothing if one doesn't exist.

**Review by zaviermiller — APPROVED** [review]

**Pull request merged**